tensorboard/plugins/text/text_plugin.py

```diff
@@ -271,44 +273,52 @@ def _async_index_impl(self):
 
   def index_impl(self):
     run_to_series = self._fetch_run_to_series_from_multiplexer()
+
     # A previous system of collecting and serving text summaries involved
     # storing the tags of text summaries within tensors.json files. See if we
     # are currently using that system. We do not want to drop support for that
     # use case.
     name = 'tensorboard_text'
     run_to_assets = self._multiplexer.PluginAssets(name)
     for run, assets in run_to_assets.items():
+      if run in run_to_series:
+        # When runs conflict, the summaries created via the new method override.
+        continue
+
       if 'tensors.json' in assets:
         tensors_json = self._multiplexer.RetrievePluginAsset(
             run, name, 'tensors.json')
         tensors = json.loads(tensors_json)
-        run_to_series[run] += tensors
+        run_to_series[run] = tensors
       else:
-        run_to_series[run] += []
+        # The mapping should contain all runs among its keys.
+        run_to_series[run] = []
+
     return run_to_series
 
   def _fetch_run_to_series_from_multiplexer(self):
-    run_to_series = collections.defaultdict(list)
     # TensorBoard is obtaining summaries related to the text plugin based on
     # SummaryMetadata stored within Value protos.
-    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
-
-    # Augment the summaries created via the deprecated (plugin asset based)
-    # method with these summaries created with the new method. When they
-    # conflict, the summaries created via the new method overrides.
-    for (run, tags) in mapping.items():
-      run_to_series[run] += tags.keys()
-    return run_to_series
+    mapping = six.iteritems(
```

Comment:
nit: strictly speaking this isn't a mapping, it's an iterable of dict items - maybe rename to "items" or conversely, move the six.iteritems() call to inside the dict comprehension.

Reply:
Done.